core/arch/arm/kernel/thread.c

```diff
@@ -358,7 +358,8 @@ static void init_regs(struct thread_ctx *thread,
 	 * Stdcalls starts in SVC mode with masked IRQ, masked Asynchronous
 	 * abort and unmasked FIQ.
 	  */
-	thread->regs.cpsr = CPSR_MODE_SVC | CPSR_I | CPSR_A;
+	thread->regs.cpsr = read_cpsr();
```

Comment:
Ok. I made a mistake. I just want to avoid endian bit not be overwritten by this function. I will close this pull request and start a new one.